namesys/publisher.go

```diff
@@ -105,6 +106,11 @@ func ipnsEntryDataForSig(e *pb.IpnsEntry) []byte {
 		[]byte{})
 }
 
+var IpnsRecordValidator = &record.ValidChecker{
+	Func: ValidateIpnsRecord,
+	Sign: true,
```

Comment:
does the `Sign:` need to be a field? could it be implemented inside the function? e.g.:

```go
type Validator func(k u.Key, data []byte) (ok bool, err error)

func sigValidator(k u.Key, data []byte) (ok bool, err error) {
  // chcek sig here 
}

// validators can compose:
func myRandomValidator(k u.Key, data []byte) (ok bool, err error) {
  if ok, err := sigValidator(k, data); err != nil || !ok {
    return ok, err
  }
  // check something else here
}


```

(nbd if it complicated things now, simple functions are pretty powerful, so maybe we can evolve towards just using functions)